GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -568,12 +581,34 @@ private void PostFetchJob(List<string> packIndexes)
                         return;
                     }
 
-                    if (!this.gitObjects.TryWriteMultiPackIndex(this.context.Tracer, this.context.Enlistment, this.context.FileSystem))
+                    this.postFetchGitProcess = new GitProcess(this.context.Enlistment);
+
+                    using (ITracer activity = this.context.Tracer.StartActivity("TryWriteMultiPackIndex", EventLevel.Informational, Keywords.Telemetry, metadata: null))
                     {
-                        this.context.Tracer.RelatedWarning(
-                            metadata: null,
-                            message: PostFetchTelemetryKey + ": Failed to generate multi-pack-index for new packfiles",
-                            keywords: Keywords.Telemetry);
+                        if (this.stopping)
+                        {
+                            this.context.Tracer.RelatedWarning(
+                                metadata: null,
+                                message: PostFetchTelemetryKey + ": Not launching 'git multi-pack-index' because the mount is stopping",
+                                keywords: Keywords.Telemetry);
+                            return;
+                        }
+
+                        GitProcess.Result result = this.postFetchGitProcess.WriteMultiPackIndex(this.context.Enlistment.GitObjectsRoot);
+
+                        if (result.HasErrors)
+                        {
+                            string message = PostFetchTelemetryKey
+                                + ": Failed to generate multi-pack-index for new packfiles"
+                                + (this.stopping ? " (expected because the process is stopping):" : ":")
+                                + result.Errors;
+
+                            this.context.Tracer.RelatedWarning(
+                                metadata: null,
+                                message: message,
+                                keywords: Keywords.Telemetry);

```

Comment:
Do we need this to go to telemetry if it's because of a shutdown? It may make more sense to restrict that to actual midx failures.